service/frontend/nexus_http_handler.go

```diff
@@ -208,6 +212,83 @@ func (h *NexusHTTPHandler) dispatchNexusTaskByService(w http.ResponseWriter, r *
 		h.writeNexusFailure(w, http.StatusNotFound, &nexus.Failure{Message: "nexus endpoints disabled"})
 		return
 	}
-	// To be implemented once the service registry is implemented.
-	w.WriteHeader(http.StatusNotImplemented)
+
+	// Limit the request body to max allowed Payload size.
+	// Content headers are transformed to Payload metadata and contribute to the Payload size as well. A separate
+	// limit is enforced on top of this in the nexusHandler.StartOperation method.
+	r.Body = http.MaxBytesReader(w, r.Body, rpc.MaxNexusAPIRequestBodyBytes)
+
+	nc := nexusContext{
+		namespaceValidationInterceptor:       h.namespaceValidationInterceptor,
+		namespaceRateLimitInterceptor:        h.namespaceRateLimitInterceptor,
+		namespaceConcurrencyLimitInterceptor: h.namespaceConcurrencyLimitInterceptor,
+		rateLimitInterceptor:                 h.rateLimitInterceptor,
+		apiName:                              configs.DispatchNexusTaskByServiceAPIName,
+	}
+
+	vars := mux.Vars(r)
+	service := commonnexus.RouteDispatchNexusTaskByService.Deserialize(vars)
+
+	serviceID, err := url.PathUnescape(*service)
+	if err != nil {
+		h.logger.Error("invalid URL", tag.Error(err))
+		h.writeNexusFailure(w, http.StatusBadRequest, &nexus.Failure{Message: "invalid URL"})
+		return
+	}
+
+	serviceInfo, err := h.incomingServiceRegistry.Get(r.Context(), serviceID)
+	if err != nil {
+		h.logger.Error("invalid Nexus incoming service ID", tag.Error(err))
+		h.writeNexusFailure(w, http.StatusBadRequest, &nexus.Failure{Message: err.Error()})
```

Comment:
It's not always a bad request, make sure to distinguish between the different errors.